src/api/controllers/db-controller.ts

```diff
@@ -828,43 +929,183 @@ export async function getTxFromDataStore(
   };
 }
 
+function parseContractWithAssociatedTx(
+  contracts: DbSmartContract[],
+  dbTxs?: DbTx[],
+  dbMempoolTx?: DbMempoolTx[]
+): Transaction[] {
+  const transactions: Transaction[] = [];
+  if (dbTxs) {
+    contracts.forEach(contract => {
+      const dbTx = dbTxs.find(tx => tx.contract_call_contract_id === contract.contract_id);
+      if (dbTx) {
+        const transaction = parseContractCallMetadata({ found: true, result: contract }, dbTx);
+        if (transaction) {
+          transactions.push(transaction as Transaction);
+        }
+      }
+    });
+  } else if (dbMempoolTx) {
+    contracts.forEach(contract => {
+      const dbMempool = dbMempoolTx.find(
+        tx => tx.contract_call_contract_id === contract.contract_id
+      );
+      if (dbMempool) {
+        const transaction = parseContractCallMetadata(
+          { found: true, result: contract },
+          undefined,
+          dbMempool
+        );
+        if (transaction) {
+          transactions.push(transaction as Transaction);
+        }
+      }
+    });
+  }
+  return transactions;
+}
+
+async function getContractCallForTxsList(
+  db: DataStore,
+  dbTxs?: DbTx[],
+  dbMempoolTxs?: DbMempoolTx[]
+): Promise<Transaction[] | undefined> {
+  // get contract call ids
+  const transactions = dbTxs ?? dbMempoolTxs ?? undefined;
```

Comment:
This function is a bit confusing. Both dbTxs and dbMempoolTxs can be passed as separate args, but only one or none will be used. Seems like this function should instead take a single arg for txs using common interface like `txs: BaseTx[]`.